src/embed.js

```diff
@@ -47,14 +35,21 @@ function embed(el, spec, opt) {
   var renderer = opt.renderer || 'canvas';
   var actions  = opt.actions !== undefined ? opt.actions : true;
 
+  var loader = opt.loader || vega.loader();
+
   // Load the visualization specification.
   if (vega.isString(spec)) {
-    return load(spec, opt, 'url', el);
+    return loader.load(spec).then(function(data) {
+      return embed(el, JSON.parse(data), opt);
+    }).catch(Promise.reject);
   }
 
   // Load Vega theme/configuration.
   if (vega.isString(opt.config)) {
-    return load(opt.config, {spec: spec, opt: opt}, 'config', el);
+    return loader.load(spec).then(function(data) {
+      opt.config = JSON.parse(data);
+      return embed(el, spec, opt);
+    }).catch(Promise.reject);
```

Comment:
I guess `loader.load(spec)` should be `loader.load(opt.config)` to load `opt.config`.